pandas/io/json/table_schema.py

```diff
@@ -306,10 +306,9 @@ def parse_table_schema(json, precise_float):
         raise NotImplementedError('table="orient" can not yet read timezone '
                                   'data')
 
-    # No ISO constructor for Timedelta as of yet, so need to raise
-    if 'timedelta64' in dtypes.values():
-        raise NotImplementedError('table="orient" can not yet read '
-                                  'ISO-formatted Timedelta data')
+    for col, dtype in dtypes.items():
```

Comment:
Is this block necessary? Assumed the subsequent `astype` call would take care of this

Reply:
So, support for iso-format timedeltas should be added to `astype` function?

Reply:
hmm, are you sure you need to do this? I believe ``.astype(dtypes)`` should handle this, and if not we should fix that.

Reply:
For now 
```python
delta = pd.Timedelta(1e9).isoformat()
pd.DataFrame([delta]).astype('timedelta64[ns]')
```
results in error.